### PR TITLE
[T3A4][T17-A3]Chia Zhe Min

### PR DESCRIPTION
**Pull request by zhems**

### CODE REVIEW

**Comment by zhems** [comment]
Ready for review

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -13,117 +13,136 @@
...
-
 /**
- * Entry point of the Address Book application.
- * Initializes the application and starts the interaction with the user.
+ * Entry point of the Address Book application. Initializes the application and
+ * starts the interaction with the user.
  */
 public class Main {
 
-    /** Version info of the program. */
```
Comment (JoelT-92) [inline src/seedu/addressbook/Main.java]: Avoid auto-formatting unrelated parts of the code. Doing so can result in unrelated changes being included in the commit. You can use sourcetree to look through changes before committing, and discard any unrelated changes.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -13,117 +13,136 @@
...
+	}
+
+	/**
+	 * Executes the command and returns the result.
+	 * 
+	 * @param command
+	 *            user command
+	 * @return result of the command
+	 */
+	private CommandResult executeCommand(Command command) throws FileDeletedException {
```
Comment (JoelT-92) [inline src/seedu/addressbook/Main.java]: It is usually not recommended to catch an exception, and throw another. You may actually just warn the user or terminate the program. Recall from the tutorial discussion which would be a better choice.

In this program, RunTimeException is not handled. This means that the program could not recover from the situation where the storage file is deleted and crash. Is this what you have planned for the program to do?

File: src/seedu/addressbook/storage/StorageFile.java [hunk 3]
```diff
@@ -95,6 +103,9 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             final Marshaller marshaller = jaxbContext.createMarshaller();
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
+            if(!Files.exists(path.getFileName())) {
```
Comment (JoelT-92) [inline src/seedu/addressbook/storage/StorageFile.java]: Recall in our discussion that FileWriter API creates the file if the file does not exist. Therefore, placing the check after that call makes it redundant.

**Comment by JoelT-92** [comment]
@zhems Some comments added. Please close the PR after reading comments. Just repeating what was mentioned in the tutorial.
